### ISSUE TITLE
long UserWindow captions (>8 chars> going to multiple lines

### ISSUE THREAD

**Issue by DarthGandalf** [kvirc/KVIrc#172]
**Reported by almostautomated on 02 Jul 2008 20:10:17 UTC**
When creating windows using $window.open and then assigning a caption with window.setcaption the captions are going to multiple lines.  Here's the script used:

```
%tstWin=$window.open(q,Test1,0,111)
window.setcaption %tstWin Test Caption
%tstWin=$window.open(q,Test2,0,111)
window.setcaption %tstWin "Test Caption2"
%tstWin=$window.open(q,Test3,0,111)
window.setcaption %tstWin "Test Caption3" <nobr><font><b>Test Caption3</nobr></font></b>
%tstWin=$window.open(q,Test4,0,111)
window.setcaption %tstWin "Test Caption4" "<nobr><font><b>Test Caption4</nobr></font></b>"
```

---

Migrated from: https://svn.kvirc.de/kvirc/ticket/172

**Comment by DarthGandalf** [comment]
**Comment by ctrlaltca on 20 Jul 2008 13:12:28**
Take care of the double quotes in your scripts, some are missing in your example :)
Anyway.. on its internal windows tipically kvirc apply a "<nobr><b><font color="from options">" to the window captions.
On $window.open kvirc was not applying any style to the user-supplied window caption. From revision 2097 it will automatically create the two formatted strings (for inactive and active caption) using kvirc theme options.
Instead, in "window.setcaption" the user can set itself those tags, so there's no need for kvirc to do any job.
This is your script, try it on kvirc rev >=2097:

#example 1 : kvirc will create the style for this window caption
%tstWin=$window.open(q,"Test Caption",0,111)
#example 2 : kvirc wont create the style for this window caption
%tstWin=$window.open(q,"",0,111)
window.setcaption %tstWin "TestCaption4" "<nobr><font><b>Test Caption4 Inactive</nobr></font></b>" "<nobr><font><b>Test Caption4 Active</nobr></font></b>"

---

Resolution: none → fixed
Status: new → closed

**Comment by DarthGandalf** [comment]
**Comment by ctrlaltca on 20 Jul 2008 13:15:00**
The example, with some wikiformatting: 

#example 1 : kvirc will create the style for this window caption<br/>
%tstWin=$window.open(q,"Test Caption",0,111)<br/>
#example 2 : kvirc wont create the style for this window caption<br/>
%tstWin=$window.open(q,"",0,111)<br/>
window.setcaption %tstWin "TestCaption?4" "<nobr><font><b>Test Caption4 Inactive</nobr></font></b>" "<nobr><font><b>Test Caption4 Active</nobr></font></b>"

**Comment by DarthGandalf** [comment]
**Comment by almostautomated on 26 Jul 2008 13:51:20**
Thank you; hopefully this fix will be in 3.4.1.  :)
